src/collision/collider/mod.rs

```diff
@@ -60,24 +214,26 @@ pub trait AnyCollider: Component + ComputeMassProperties {
         feature = "2d",
         doc = "\n\nThe rotation is counterclockwise and in radians."
     )]
-    fn swept_aabb(
+    fn swept_aabb_with_context(
         &self,
+        context: AabbContext<Self::Context>,
         start_position: Vector,
         start_rotation: impl Into<Rotation>,
         end_position: Vector,
         end_rotation: impl Into<Rotation>,
     ) -> ColliderAabb {
-        self.aabb(start_position, start_rotation)
-            .merged(self.aabb(end_position, end_rotation))
+        self.aabb_with_context(context.clone(), start_position, start_rotation)
+            .merged(self.aabb_with_context(context, end_position, end_rotation))
     }
 
     /// Computes all [`ContactManifold`]s between two colliders.
     ///
     /// Returns an empty vector if the colliders are separated by a distance greater than `prediction_distance`
     /// or if the given shapes are invalid.
-    fn contact_manifolds(
+    fn contact_manifolds_with_context(
         &self,
         other: &Self,
+        context: ContactManifoldContext<Self::Context>,
         position1: Vector,
         rotation1: impl Into<Rotation>,
         position2: Vector,
```

Comment:
Same as the AABB methods, I feel like the context should be at the end. In this case I also consider the positions and rotations to be kinda "part of" the shape definition, so it's natural to have them next to each other, while the context isn't strictly related to the shapes themselves.